### PR TITLE
update components & models to v1beta1 schema definitions

### PR DESCRIPTION
**Pull request by code-sleuth**
**Description**
Update components & models to v1beta1 schema definitions

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: adapter/oam.go [hunk 1]
```diff
@@ -85,7 +84,7 @@ func CreateComponents(scfg StaticCompConfig) error {
 	}
```
Comment (MUzairS15) [inline adapter/oam.go:84]: @code-sleuth Where is the schema being assigned?

**Review by MUzairS15 — COMMENTED** [review]
@code-sleuth Thanks for volunteering 👍, but it might be that this is too much for your initial issues.
I will push some commits to this branch.

**Comment by MUzairS15** [comment]
https://docs.meshery.io/search/?q=adapter
There are few make targets that help you regenerate comps, 
Use that to generate comps, start an adapter, and verify if comps are registered with the server.
To verify if comps are registered use `Registry Browser` in Meshery UI under `Settings` page.
OR use [models API](https://docs.meshery.io/reference/rest-apis#api-meshmodel-components-register).
The endpoint listed [here](https://docs.meshery.io/reference/rest-apis#api-meshmodel-components-register) is what adapters use to register comps.

**Comment by MUzairS15** [comment]
[Slide 42](https://docs.google.com/presentation/d/1SQMfyu5shjpGKlYONdVzOtd7UYTgLWBcgUvHMLCZ2tY/edit#slide=id.gb9870e7157_1_0) inside Meshery Architecture deck captures the flow for adaptes.

File: adapter/oam.go [hunk 2]
```diff
@@ -107,21 +107,21 @@ func convertOAMtoMeshmodel(def []byte, schema string, isCore bool, meshmodelname
...
-		c.APIVersion = oamdef.APIVersion
-		c.Kind = oamdef.ObjectMeta.Name
+		c.Component.TypeMeta.Version = oamdef.APIVersion
+		c.Component.TypeMeta.Kind = oamdef.ObjectMeta.Name
 		c.Model.Version = oamdef.Spec.Metadata["version"]
 		c.Model.Name = meshmodelname
 	} else {
-		c.APIVersion = oamdef.Spec.Metadata["k8sAPIVersion"]
-		c.Kind = oamdef.Spec.Metadata["k8sKind"]
+		c.Component.TypeMeta.Version = oamdef.Spec.Metadata["k8sAPIVersion"]
```
Comment (MUzairS15) [inline adapter/oam.go:115]: Ensure `schemaVersion` and `defVersion` are assigned.

**Review by MUzairS15 — COMMENTED** [review]
@code-sleuth Were you able to generate and register these components?

**Comment by MUzairS15** [comment]
@code-sleuth any updates?

**Comment by code-sleuth** [comment]
@MUzairS15 this is ready to ship

File: adapter/oam.go (hunk 1 above)
Comment (code-sleuth) [reply adapter/oam.go]: https://github.com/meshery/meshery-adapter-library/blob/8f581b2369730d0a2589d068e5066fa09d8a7808/adapter/oam.go#L124

**Comment by code-sleuth** [comment]
@MUzairS15 friendly ping

**Comment by leecalcote** [comment]
@lekaf974, I wonder if you might be inclined to offer a review there.

**Comment by lekaf974** [comment]
Not being a lot involve in meshkit components until now but a good opportunity to learn so I'll take a look after make some research

**Pull request closed without merging**